core/templates/dev/head/pages/exploration_editor/editor_tab/StateHints.js

```diff
@@ -157,8 +187,17 @@ oppia.controller('StateHints', [
           }
         ]
       }).result.then(function() {
-        stateHintsService.displayed.splice(index, 1);
-        stateHintsService.saveDisplayedValue();
+        if (stateSolutionService.displayed) {
+          if (index > 0) {
+            stateHintsService.displayed.splice(index, 1);
+            stateHintsService.saveDisplayedValue();
+          } else {
+            openDeleteLastHintModal();
```

Comment:
Not sure I understand the logic here. What if you have more than one hint but delete the first one?